### ISSUE TITLE
DocumentClient with permission feed authentication does not work as it should.

### ISSUE THREAD

**Issue by ankurgupta4891** [Azure/azure-documentdb-node#178]
The following snippet is of DocumentClient.js client. When client is initialized with permissions array, it builds the resourceTokens object for resourceId and token mappings. 
```
if (auth.permissionFeed) {
	this.resourceTokens = {};
	for (var i = 0; i < auth.permissionFeed.length; i++) {
		var resourceParts = auth.permissionFeed[i].resource.split("/");
		var rid = resourceParts[resourceParts.length - 1];
		this.resourceTokens[rid] = auth.permissionFeed[i]._token;
	}
}
```
Since resource url in permission object ends with "/" (see example below). Because of it, `var rid = resourceParts[resourceParts.length - 1];` returns empty string. In next line, this.resourceTokens object will have empty string property instead of having rid as property and ends up having only last permission's token in resourceTokens object.  

```
_etag:""00008c01-0000-0000-0000-58e532230000""
_rid:"ooUSAGBSQwCKteUDHY4mAB=="
_self:"dbs/ooUSAW==/users/ooUSAGBSQwV=/permissions/ooUSAGBSQwCKteUDHY4mAB==/"
_token:"type=resource&ver=1&sig=nEX/ym6+McLXjnftlr2FBQ==;DaD+rh2vqDsctFO9NKJZxYZh1J9FaAc/L3sgqLo3LVcGT5xvUY7xzalWFbVTLWjY/wCAAGHA9HIIQJzweAbwWgKRpZT/2FfRW4JE8L7ShucEGQPlQ0MsUFnU1nZLyHE5FEFDiTFQXi3YpXbtmuZknbfzD8P62BBfYxE9899LevyujTTN03g0RohWAKHLImmfgfZHwVEpA4VnxrpxLgB0wBqB9rDHvMjm5iLoWMqVdTtGCoGdvsHauuAhDhP5q9xF;"
_ts:1491415583
id:"perm1"
permissionMode:"All"
resource:"dbs/ooUSAA==/colls/ooUSALK3-QE=/"
```

**Comment by mehdijazayeri3000** [comment]
Could you please share your code with us so that we can reproduce the problem?

**Comment by moderakh** [comment]
fixed in 1.12.2
